efficientdet/keras/train.py

```diff
@@ -0,0 +1,210 @@
+
+FLAGS = flags.FLAGS
+
+
+def main(_):
+  # Parse and override hparams
+  config = hparams_config.get_detection_config(FLAGS.model_name)
+  config.override(FLAGS.hparams)
+  if FLAGS.num_epochs:  # NOTE: remove this flag after updating all docs.
+    config.num_epochs = FLAGS.num_epochs
+
+  # Parse image size in case it is in string format.
+  config.image_size = utils.parse_image_size(config.image_size)
+
+  if FLAGS.use_xla and FLAGS.strategy != 'tpu':
+    tf.config.optimizer.set_jit(True)
+    tf.config.experimental.set_memory_growth(True)
+
+  if FLAGS.debug:
+    tf.config.experimental_run_functions_eagerly(True)
+    tf.debugging.set_log_device_placement(True)
+    tf.random.set_seed(111111)
+    logging.set_verbosity(logging.DEBUG)
+
+  if FLAGS.strategy == 'tpu':
+    tpu_cluster_resolver = tf.distribute.cluster_resolver.TPUClusterResolver(
+        FLAGS.tpu, zone=FLAGS.tpu_zone, project=FLAGS.gcp_project)
+    tf.config.experimental_connect_to_cluster(tpu_cluster_resolver)
+    tf.tpu.experimental.initialize_tpu_system(tpu_cluster_resolver)
+    ds_strategy = tf.distribute.TPUStrategy(tpu_cluster_resolver)
+    logging.info("All devices: ", tf.config.list_logical_devices('TPU'))
+  elif FLAGS.strategy == 'gpus':
+    ds_strategy = tf.distribute.MirroredStrategy()
+    logging.info("All devices: ", tf.config.list_physical_devices('GPU'))
+  else:
+    if tf.config.list_physical_devices('GPU'):
+      ds_strategy = tf.distribute.OneDeviceStrategy("device:GPU:0")
+    else:
+      ds_strategy = tf.distribute.OneDeviceStrategy("device:CPU:0")
+
+  # Check data path
+  if FLAGS.mode in ('train',
+                    'train_and_eval') and FLAGS.training_file_pattern is None:
+    raise RuntimeError('You must specify --training_file_pattern for training.')
+  if FLAGS.mode in ('eval', 'train_and_eval'):
+    if FLAGS.validation_file_pattern is None:
+      raise RuntimeError('You must specify --validation_file_pattern '
+                         'for evaluation.')
+
+  params = dict(config.as_dict(),
+                model_name=FLAGS.model_name,
+                iterations_per_loop=FLAGS.iterations_per_loop,
+                model_dir=FLAGS.model_dir,
+                num_examples_per_epoch=FLAGS.num_examples_per_epoch,
+                strategy=FLAGS.strategy,
+                batch_size=FLAGS.batch_size,
+                num_shards=FLAGS.num_cores,
+                val_json_file=FLAGS.val_json_file,
+                testdev_dir=FLAGS.testdev_dir,
+                mode=FLAGS.mode)
+
+  # set mixed precision policy by keras api which will generate mixed precision model
+  precision = utils.get_precision(params['strategy'], params['mixed_precision'])
+  policy = tf.keras.mixed_precision.experimental.Policy(precision)
+  tf.keras.mixed_precision.experimental.set_policy(policy)
+
+  def get_dataset(is_training, params):
+    global_batch_size = FLAGS.batch_size
+    batch_size = global_batch_size // ds_strategy.num_replicas_in_sync
```

Comment:
Where is this batch_size used? 
Looks like it should be params['batch_size'] = ...?